provider/pkg/autonaming/application.go

```diff
@@ -70,6 +91,19 @@ func getDefaultName(
 	if autoNamingSpec.MaxLength > 0 {
 		left := autoNamingSpec.MaxLength - len(prefix) - namingTrivia.Length()
 
+		if left <= 0 && autoTrim {
+			autoTrimMaxLength := autoNamingSpec.MaxLength - namingTrivia.Length() - randomSuffixMinLength
+			if autoTrimMaxLength <= 0 {
+				return resource.PropertyValue{}, fmt.Errorf("failed to auto-generate value for %[1]q."+
```

Comment:
So surprised to see Fortran-style 1-based indices here but I think this checks out. This is right.